reputation/reputation.go

```diff
@@ -0,0 +1,576 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package reputation
+
+import (
+	"bytes"
+	"database/sql"
+	"fmt"
+	"log"
+
+	"github.com/google/uuid"
+	"github.com/zeebo/errs"
+	// import of sqlite3 for side effects
+	_ "github.com/mattn/go-sqlite3"
+)
+
+// StartDBError is an error class for errors related to the reputation package
+var StartDBError = errs.Class("reputation start sqlite3 error")
+
+// CreateTableError is an error class for errors related to the reputation package
+var CreateTableError = errs.Class("reputation table creation error")
+
+// InsertError is an error class for errors related to the reputation package
+var InsertError = errs.Class("reputation insertion error")
+
+// SelectError is an error class for errors related to the reputation package
+var SelectError = errs.Class("reputation selection error")
+
+// IterError is an error class for errors related to the reputation package
+var IterError = errs.Class("reputation iteration error")
+
+// DeleteError is an error class for errors related to the reputation package
+var DeleteError = errs.Class("reputation deletion error")
+
+// NodeReputationRecord is the Data type for Rows in Reputation table
+type NodeReputationRecord struct {
+	name               string
+	timestamp          string
+	uptime             int
+	auditSuccess       int
+	auditFail          int
+	latency            int
+	amountOfDataStored int
+	falseClaims        int
+	shardsModified     int
+}
+
+// startDB starts a sqlite3 database from the file path parameter
+func startDB(filePath string) (*sql.DB, error) {
+	db, err := sql.Open("sqlite3", filePath)
+	if err != nil {
+		log.Printf("%q\n", err)
+		return nil, StartDBError.Wrap(err)
+	}
+
+	return db, nil
+}
+
+// createTable creates a table in sqlite3 based on the create table string parameter
+func createTable(createStmt string, db *sql.DB) error {
+	_, err := db.Exec(createStmt)
+	if err != nil {
+		log.Printf("%q: %s\n", err, createStmt)
+		return InsertError.Wrap(err)
+	}
+	return nil
+}
+
+// createNamedRow creates a reputation row struct with a name field base on the name parameter
+func createNamedRow(seed int, name string) NodeReputationRecord {
+	return NodeReputationRecord{
+		name:               name,
+		timestamp:          "",
+		uptime:             seed,
+		auditSuccess:       seed,
+		auditFail:          seed,
+		latency:            seed,
+		amountOfDataStored: seed,
+		falseClaims:        seed,
+		shardsModified:     0,
+	}
+}
+
+// createRandRows create a slice of reputation row with random data with the number of rows based on the max row parameter
+func createRandRows(numRows int) []NodeReputationRecord {
+	res := make([]NodeReputationRecord, 0, numRows)
+
+	for i := 0; i <= numRows; i++ {
+		res = append(res, createNamedRow(i, uuid.New().String()))
```

Comment:
so numRow 100 starts off with a `falseClaims` count of 100, but numRow 1 starts off with a `falseClaims` count of 1?

also, since `i <= numRows` and not just `i < numRows`, doesn't this mean there are `1+numRows` rows?

last, uuids are actually not nearly as good as just straight up random bytes from a cryptographic entropy source. the `crypto/rand` package combined with `encoding/base32` or `encoding/base64`  is a good choice.

i think you're still working on this, so super sorry if this feedback is too early. should i wait until this is no longer in WIP state?